src/text/line.rs

```diff
@@ -208,8 +199,6 @@ impl<'a> Line<'a> {
     ///
     /// # Examples
     ///
-    /// Any newlines in the content are removed.
```

Comment:
A Line is something without newlines. (Text has multiple Lines)

The \n could be removed and end up in a single span, but that would result in cloned data and less often borrowed one. That would likely end up with more allocations as before.

Maybe the implementation of Line drawing could ensure to ignore `\n` resulting in the same behaviour?

Reply:
Yeah, I guess my point is that line drawing already seems to ignore lines. I can check again, but it didn't seem like a line with newlines rendered differently with this change.

Reply:
Yup, just checked again and sticking a `\n` in a `Line::raw` doesn't do anything with or without this change.

Reply:
The rendering of `\n` in Spans is prevented by this line: https://github.com/ratatui-org/ratatui/blob/b344f95b7cfc3b90221c84cfef8afcf6944bafc1/src/text/span.rs#L274

It's also a control character so set_stringn wouldn't render this either:
https://github.com/ratatui-org/ratatui/blob/b344f95b7cfc3b90221c84cfef8afcf6944bafc1/src/buffer/buffer.rs#L213

But it looks like `Span::render_ref` and `set_stringn` diverged. (Which also results in #1226 being effectless for Spans, so the bug is still there for Spans). Personally I would just remove the logic in Spans and use set_stringn under the hood there, but it seems to contain some strangely complex code from a bugfix #1165 fixing #1160. That's either missing from set_stringn or not a problem at all there?

Tldr… lost way too much time trying to understand that part of the ratatui code…
But they, that sparked #1257 and #1258.

Reply:
> Tldr… lost way too much time trying to understand that part of the ratatui code…
But they, that sparked #1257 and #1258.

Oof, but also hell yeah! I'm a bit lost—where does that leave this PR? Sounds like you agree that newlines don't need to be removed anymore?